src/chttpd_auth.erl

```diff
@@ -12,15 +12,70 @@
 
 -module(chttpd_auth).
 
+-export([authenticate/2]).
+-export([authorize/2]).
+
 -export([default_authentication_handler/1]).
 -export([cookie_authentication_handler/1]).
+-export([local_admin_handler/1]).
+
 -export([handle_session_req/1]).
 
+-include_lib("couch/include/couch_db.hrl").
+
+-define(SERVICE_ID, chttpd_auth).
+
+
+%% ------------------------------------------------------------------
+%% API Function Definitions
+%% ------------------------------------------------------------------
+
+authenticate(HttpReq, Default) ->
+    maybe_handle(authenticate, [HttpReq], Default).
+
+authorize(HttpReq, Default) ->
+    maybe_handle(authorize, [HttpReq], Default).
+
+
+%% ------------------------------------------------------------------
+%% Default callbacks
+%% ------------------------------------------------------------------
+
 default_authentication_handler(Req) ->
     couch_httpd_auth:default_authentication_handler(Req, chttpd_auth_cache).
 
 cookie_authentication_handler(Req) ->
     couch_httpd_auth:cookie_authentication_handler(Req, chttpd_auth_cache).
 
+local_admin_handler(Req) ->
+    case config:get("chttpd", "require_valid_user", "false") of
```

Comment:
So, since you touched this line, bikeshedding question: we have same option defined in "couch_http_auth" [\[1\]](https://github.com/apache/couchdb-couch/blob/master/src/couch_httpd_auth.erl#L106) [\[2\]](https://github.com/apache/couchdb-couch/blob/master/src/couch_httpd.erl#L409)  section. Shouldn't it be used here as well to avoid having same things duplicated?